packages/zapp/console/src/components/Launch/LaunchForm/test/ResumeSignalForm.test.tsx

```diff
@@ -0,0 +1,195 @@
+import { ThemeProvider } from '@material-ui/styles';
+import { fireEvent, queryAllByRole, render, waitFor } from '@testing-library/react';
+import { APIContext } from 'components/data/apiContext';
+import { mockAPIContextValue } from 'components/data/__mocks__/apiContext';
+import { muiTheme } from 'components/Theme/muiTheme';
+import { SimpleType } from 'models/Common/types';
+import { resumeSignalNode } from 'models/Execution/api';
+import * as React from 'react';
+import { NodeExecutionsByIdContext } from 'components/Executions/contexts';
+import { dateToTimestamp } from 'common/utils';
+import { NodeExecutionPhase } from 'models/Execution/enums';
+import { createTestQueryClient } from 'test/utils';
+import { mockWorkflowId } from 'mocks/data/fixtures/types';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import { Core } from 'flyteidl';
+import { CompiledNode } from 'models/Node/types';
+import { CompiledWorkflowClosure } from 'models/Workflow/types';
+import { NodeExecutionDetailsContext } from 'components/Executions/contextProvider/NodeExecutionDetails';
+import { signalInputName } from './constants';
+import { ResumeSignalForm } from '../ResumeSignalForm';
+
+const mockNodeExecutionId = 'n0';
+const mockNodeId = 'node0';
+
+const mockNodeExecutionsById = {
+  [mockNodeExecutionId]: {
+    closure: {
+      createdAt: dateToTimestamp(new Date()),
+      outputUri: '',
+      phase: NodeExecutionPhase.UNDEFINED,
+    },
+    id: {
+      executionId: { domain: 'domain', name: 'name', project: 'project' },
+      nodeId: mockNodeId,
+    },
+    inputUri: '',
+    scopedId: mockNodeExecutionId,
+  },
+};
+
+const createMockCompiledWorkflowClosure = (nodes: CompiledNode[]): CompiledWorkflowClosure => ({
+  primary: {
+    connections: {
+      downstream: {},
+      upstream: {},
+    },
+    template: {
+      id: mockWorkflowId,
+      nodes,
+    },
+  },
+  tasks: [],
+});
+
+const createMockCompiledNode = (type: Core.ILiteralType | null): CompiledNode => ({
+  id: mockNodeExecutionId,
+  metadata: {
+    name: 'my-signal-name',
+    timeout: '3600s',
+    retries: {},
+  },
+  upstreamNodeIds: [],
+  gateNode: {
+    signal: {
+      signalId: 'my-signal-name',
+      type,
+      outputVariableName: 'o0',
+    },
+  },
+});
+
+describe('ResumeSignalForm', () => {
+  let onClose: jest.Mock;
+  let queryClient: QueryClient;
+  let mockResumeSignalNode: jest.Mock<ReturnType<typeof resumeSignalNode>>;
+
+  beforeEach(() => {
+    onClose = jest.fn();
+    queryClient = createTestQueryClient();
+  });
+
+  const renderForm = (type: Core.ILiteralType | null) => {
```

Comment:
isn't it simpler to make type optional `type?: Core.ILiteralType` and then call `renderForm()` without params?